src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -216,16 +211,86 @@ public static void AddExceptionInformation(in ReadStack state, in Utf8JsonReader
             }
         }
 
+        public static NotSupportedException GetNotSupportedException(in ReadStack state, in Utf8JsonReader reader, NotSupportedException ex)
```

Comment:
```suggestion
        [DoesNotReturn]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public static void GetNotSupportedException(in ReadStack state, in Utf8JsonReader reader, NotSupportedException ex)
```

Reply:
As mentioned earlier, we need to use plain "throw" statement to re-throw the current exception. We don't want to throw the same exception instance since that will create a new call stack unnecessarily. So this method can't be `[DoesNotReturn]`

Reply:
Update: added check for "Path" to caller and changed method to add `[DoesNotReturn]`. We still keep the original "throw" in the caller's catch statement in order to preserve the original call stack.